### DISCUSSION TITLE
IQuote not found?

### DISCUSSION THREAD

**Discussion by Jenscaasen** [DaveSkender/Stock.Indicators#203] [Q&A] [answered]
Hi,
this page suggests to implement "IQuote" rather than creating a child object of Quote:
https://dotnet.stockindicators.dev/guide#using-custom-quote-classes

But no matter what i include, IQuote can not be resolved using the Nuget Package.

My goal here is to try out the performance of the Lib when using IQuote, as i currently loose a lot of processor time converting my custom Quote Class back to "Quote"
Do i miss a package?

**Comment by DaveSkender** [comment, marked as answer]
You cannot modify `IQuote` directly, rather you'd need to add it to your pre-existing class equivalent of `Quote`.  The code in your app should look something like this:
```csharp
using Skender.Stock.Indicators;

[..]

public class MyCustomQuote : IQuote
{
    // required base properties
    public DateTime Date { get; set; }
    public decimal Open { get; set; }
    public decimal High { get; set; }
    public decimal Low { get; set; }
    public decimal Close { get; set; }
    public decimal Volume { get; set; }

    // custom properties
    public int MyOtherProperty { get; set; }
}
```
If you're getting [CS2046 'IQuote' could not be found](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0246), then I'd check to make sure you've 1)  supplied the `using` statement (see above) and 2) specified -- and downloaded -- the latest version of the library in your project (see below).
```xml
<ItemGroup>
    <PackageReference Include="Skender.Stock.Indicators" Version="1.2.0" />   <!-- or newer -->
</ItemGroup>
```
If you've implemented a custom Quote class with the exact same `Quote` name as the library, you can also try using clarifying approaches like using prefixes like `Skender.Stock.Indicators.IQuote` or an [alias directive](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-directive) to clarify the name duplication.

cc @MithrilMan, thoughts?
